### PR TITLE
Drop support for legacy graph drawing, add support for synthetic nodes for elided revisions

### PR DESCRIPTION
**Pull request by martinvonz**
I hope to enable this by default, so please enable it and see how you like it

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/graphlog.rs [hunk 1]
```diff
@@ -138,894 +138,13 @@ pub fn get_graphlog<'a, K: Clone + Eq + Hash + 'a>(
...
-        | |\ \ \  with
-        | | : ~/  many
-        | ~/ /    lines
-        |/ /      of
-        |/        text [width=10]
-        o node 1 [width=2]
-        "###);
-
-        Ok(())
+        _ => SaplingGraphLog::create(builder.build_box_drawing(), formatter, "◉"),
```
Comment (yuja) [inline cli/src/graphlog.rs]: nit: insert `// "curved"` as inline comment? It'll help grepping.

File: cli/src/commands/log.rs [hunk 2]
```diff
@@ -104,6 +105,11 @@ pub(crate) fn cmd_log(
         Some(value) => value.to_string(),
         None => command.settings().config().get_string("templates.log")?,
     };
+    let use_elided_nodes = command
+        .settings()
+        .config()
+        .get_bool("log.synthetic-elided-nodes")
```
Comment (yuja) [inline cli/src/commands/log.rs:111]: need to update `config-schema.json`.

Optionally, the default can be defined in `config/misc.toml` and error out on missing/invalid value.

File: cli/src/commands/log.rs [hunk 3]
```diff
@@ -127,19 +134,36 @@ pub(crate) fn cmd_log(
...
+                        RevsetGraphEdgeType::Indirect => {
+                            if use_elided_nodes {
+                                let mut synthetic_commit_id_bytes = edge.target.to_bytes();
+                                // Let's hope this commit id doesn't exist
+                                synthetic_commit_id_bytes.push(0);
+                                let synthetic_commit_id = CommitId::new(synthetic_commit_id_bytes);
+                                elided.push((synthetic_commit_id.clone(), edge.target));
+                                graphlog_edges.push(Edge::Present {
+                                    direct: true,
+                                    target: synthetic_commit_id,
```
Comment (yuja) [inline cli/src/commands/log.rs:158]: Perhaps, synthetic target could be `Real(CommitId) | Synthetic(CommitId)` enum, but that would require extra cloning. I'm fine with either way.

**Review by yuja — APPROVED** [review]
Nice.

File: cli/src/commands/log.rs (hunk 3 above)
Comment (martinvonz) [reply cli/src/commands/log.rs]: Oh, that reminds me that I meant to try switching to a `(CommitId, bool)` pair as target instead. I've done that. Let me know if you prefer an enum.

File: cli/src/commands/log.rs [hunk 4]
```diff
@@ -166,18 +186,38 @@ pub(crate) fn cmd_log(
...
                     &graphlog_edges,
                     node_symbol,
                     &String::from_utf8_lossy(&buffer),
                 )?;
+                for elided_target in elided_targets {
+                    let elided_key = (elided_target, true);
+                    let real_key = (elided_key.0.clone(), false);
+                    let edges = [Edge::Present {
+                        direct: true,
+                        target: real_key.clone(),
```
Comment (yuja) [inline cli/src/commands/log.rs:206]: nit: redundant `.clone()`

**Review by yuja — APPROVED** [review]
Thanks!

**Pull request merged**